src/detections/hooks/use-get-detections.js

```diff
@@ -1,18 +1,15 @@
-/* eslint-disable */
-import { useEffect } from 'react'
+import { useCallback } from 'react'
 import { useDispatch } from 'react-redux'
 
 import { actions } from 'detections/fetch/slices/get'
 import { useContext } from 'detections/context'
 
 export const useGetDetections = () => {
   const dispatch = useDispatch()
-  const { attempt, reset } = actions
+  const { attempt } = actions
   const { pageIndex, pageSize } = useContext()
 
-  useEffect(() => {
+  return useCallback(() => {
     dispatch(attempt({ pagination: { pageIndex, pageSize } }))
-    dispatch(reset({ pagination: { pageIndex, pageSize } }))
-  }, [dispatch, attempt, pageIndex, pageSize, reset])
+  }, [dispatch])
```

Comment:
Shouldn't we leave in the `pageIndex` and `pageSize` variables as dependencies to this `useCallback`? After all, the table's pagination depends on it. :smile:

Reply:
done